### ISSUE TITLE
Non-root pipeline images: build public certs directly into our worker binary

### ISSUE THREAD

**Issue by msteffen** [pachyderm/pachyderm#3863]
Currently the pachyderm worker binary copies a collection of included public SSL certs into `/etc/ssl` in its host container so that it can talk to public object store providers (e.g. during egress or during merges). However, when pipelines use docker images that don't run as root by default, the worker binary doesn't have the necessary permissions to do that copy.

Instead, the worker binary could have a collection of public certs built directly into the binary, and use those for communicating with object store. Then it wouldn't need to copy certs into `/etc` and would just work out of the box.

**Comment by gabrielgrant** [comment]
IIUC this issue of stale certs isn't really made any worse if the certs are baked into the binary, is it? Right now the certs are baked into the worker docker image, which, while it could theoretically be updated, is effectively static (at least as much so as the binary it contains). Am i missing something here?

**Comment by msteffen** [comment]
Nope, that's true (stale certs aren't any more of a problem if we build them into the binary than they are now)—hadn't thought that through all the way
